### PR TITLE
add tests for app: run, build, deploy, and undeploy (Extension Registry release)

### PR DESCRIPTION
**Pull request by shazron**
This PR's base branch is `release-ext-reg`

## How Has This Been Tested?

unit tests all pass:
- [X] app:run
- [X]  app:build
- [X] app:deploy
- [X] app:undeploy

code coverage 100%:
- [X] app:run
- [X]  app:build
- [X] app:deploy
- [X] app:undeploy

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: test/commands/app/build.test.js [hunk 1]
```diff
@@ -123,14 +126,21 @@ const sampleAppConfig = {
   actions: { src: '/actions', dist: '/dist/actions', devRemote: false },
   root: path.resolve('test/__fixtures__/sample-app')
 }
+
+const createAppConfig = (aioConfig = {}, appFixtureName = 'legacy-app') => {
```
Comment (moritzraho) [inline test/commands/app/build.test.js:130]: I am a bit worried that most of the tests use the legacy-app mock test, which is a corner case, but we can modify this later if needed

**Review by moritzraho — APPROVED** [review]
LGTM, great progress !

File: test/commands/app/build.test.js (hunk 1 above)
Comment (shazron) [reply test/commands/app/build.test.js:130]: yeah, it was to make the existing unit-tests work without much modification. we can augment each test by running the command again with the exc-app-nui config
